src/emissions-workspace/emissions-checks.service.ts

```diff
@@ -16,17 +17,67 @@ export class EmissionsChecksService {
     }
   }
 
-  runChecks(payload: EmissionsImportDTO)  {
+  runChecks(payload: EmissionsImportDTO): string[] {
     this.logger.info('Running Emissions Import Checks');
 
     const errorList: string[] = [];
-    const weeklyTestSummaryCheckErrors = this.weeklyTestSummaryCheckService.runChecks(payload);
+    const weeklyTestSummaryCheckErrors = this.weeklyTestSummaryCheckService.runChecks(
+      payload,
+    );
+    const invalidDatesCheckErrors = this.invalidDatesCheck(payload);
 
-    errorList.push(...weeklyTestSummaryCheckErrors);
+    errorList.push(...weeklyTestSummaryCheckErrors, ...invalidDatesCheckErrors);
 
     this.throwIfErrors(errorList);
     this.logger.info('Completed Emissions Import Checks');
 
     return errorList;
   }
+
+  // IMPORT-23
+  invalidDatesCheck(payload: EmissionsImportDTO): string[] {
+    let earliestDate: number = undefined;
+    let latestDate: number = undefined;
+
+    const dateCheck = (date: Date) => {
+      const year = date.getFullYear();
+      const quarter = Math.floor(date.getMonth() / 3 + 1);
+      const combo = Number(`${year}${quarter}`);
+
+      if (typeof earliestDate === 'undefined' || combo < earliestDate) {
+        earliestDate = combo;
+      }
+
+      if (typeof latestDate === 'undefined' || combo > latestDate) {
+        latestDate = combo;
+      }
+    };
+
+    payload.dailyEmissionData?.forEach(datum => {
+      dateCheck(datum.date);
+    });
+
+    payload.dailyTestSummaryData?.forEach(datum => {
+      dateCheck(datum.date);
+    });
+
+    payload.hourlyOperatingData?.forEach(datum => {
+      dateCheck(datum.date);
+    });
+
+    payload.sorbentTrapData?.forEach(datum => {
+      dateCheck(datum.beginDate);
```

Comment:
SorbentTrab has both a beginDate and endDate. Would we need another `dateCheck(datum.endDate)` in this loop?